sqlserver/datadog_checks/sqlserver/statements.py

```diff
@@ -51,34 +51,58 @@
 with qstats as (
     select TOP {limit} query_hash, query_plan_hash, last_execution_time, plan_handle,
            (select value from sys.dm_exec_plan_attributes(plan_handle) where attribute = 'dbid') as dbid,
+            statement_start_offset,
+            statement_end_offset,
            {query_metrics_columns}
     from sys.dm_exec_query_stats
     where last_execution_time > dateadd(second, -?, getdate())
 ),
 qstats_aggr as (
     select query_hash, query_plan_hash, CAST(S.dbid as int) as dbid,
        D.name as database_name, max(plan_handle) as plan_handle,
+       statement_start_offset,
+       statement_end_offset,
     {query_metrics_column_sums}
     from qstats S
     left join sys.databases D on S.dbid = D.database_id
-    group by query_hash, query_plan_hash, S.dbid, D.name
+    group by query_hash, query_plan_hash, S.dbid, D.name, statement_start_offset, statement_end_offset
 )
-select text, encrypted as is_encrypted, * from qstats_aggr
-    cross apply sys.dm_exec_sql_text(plan_handle)
+select
+    SUBSTRING(text, (statement_start_offset / 2) + 1,
```

Comment:
Since we're reading out the full text anyway, would it make sense to read out `statement_start_offset` and `statement_end_offset` and do the substring in the agent instead of having the DB do it? It would be slightly less work for the DB and less data to send out. I'm not sure if it would make much of a difference, but since we've had performance issues with these queries before I'd err on the side of keeping them as lean as possible.

Reply:
yeah, I'd imagine it wouldn't make much of a difference but I can update it to do that

Reply:
hmmm I can't seem to get this working locally. I updated the rest of the PR with your feedback, but getting mixed results when using this func

```
    def _get_statement_text(row):
        orig_text = row['text']
        if 'statement_start_offset' not in row or 'statement_end_offset' not in row:
            return orig_text
        # sqlserver does integer divison for this calclation. See this doc:
        # https://docs.microsoft.com/en-us/sql/t-sql/language-elements/divide-transact-sql?view=sql-server-ver16
        start = (row['statement_start_offset'] // 2) + 1
        end_offset = row['statement_end_offset']
        if end_offset == -1:
            end_offset = len(row['text'])
        end = (end_offset - row['statement_start_offset'] // 2) + 1
        return orig_text[start:end]
 ```
 
 I must be doing something wrong, but its not obvious what and I don't have time to figure it out before I leave. Maybe this change can be a followup optimization?